### PR TITLE
Core/Battlegrounds: Fix log errors about creatures not found

### PR DESCRIPTION
**Pull request by grimgravy**
##### CHANGES PROPOSED:

-  Move "Battleground::GetBGCreature: creature not found for BG" error log to "GetBgMap" level when the caller expects no creatures to be found, like on Battleground startup.
-  remove wrong DB errors.
It is normal to have events applicable only to certain instance difficulties.

###### ISSUES ADDRESSED:
I also saw the problem not only in the alterac valley, also on the island of conquest, but able to fix the repair based on trinitycore

Closes [#1496](https://github.com/azerothcore/azerothcore-wotlk/issues/1496)

##### TESTS PERFORMED:
windows 7 operating system

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@grimgravy can you please provide instructions for testers? I guess we need to make sure that BGs still work fine. Maybe you can provide a list of what can be affected by this change i.e. what testers should double check

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -4168,13 +4168,11 @@ void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTrigger c
 {
     if (e.empty())
     {
-#if defined(ENABLE_EXTRAS) && defined(ENABLE_EXTRA_LOGS)
```
Comment (Stoabrogga) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]: I think this should not be deleted as it controls debug log entries.

Comment (grimgravy) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: I need to do this so travis can release the checks

Comment (Stoabrogga) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: I don't understand, what do you mean by that?

Comment (Stoabrogga) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: Ok, I understand. Parameter "at" is unused if extra logging is disabled and the compiler complains about this. But this has to be resolved in another way, not by deleting the preprocessor directives, as you will get debug messages even when disabling extra logs.

Comment (grimgravy) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: So, I will revert to original changes.

Comment (Stoabrogga) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: You could add
```(void)at;```
at the beginning of the function. This way the compiler sees that the parameter is referenced and won't complain any more.

Comment (grimgravy) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: Thanks for the help.

**Comment by FrancescoBorzi** [comment]
How to test this?

**Comment by grimgravy** [comment]
@FrancescoBorzi, you can rest easy, there is no problem in the operation of bg. For if you hover problems post here

**Comment by FrancescoBorzi** [comment]
@grimgravy sorry I don't understand

**Comment by grimgravy** [comment]
@FrancescoBorzi, bgs are working normally, do not worry if there are problems post here.
Sorry, I use to translate the English the google translator, sometimes give problem to translate

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 2]
```diff
@@ -4197,10 +4197,6 @@ void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTrigger c
         }
         mEvents.push_back((*i));//NOTE: 'world(0)' events still get processed in ANY instance mode
     }
-    if (mEvents.empty() && obj)
-        sLog->outErrorDb("SmartScript: Entry %u has events but no events added to list because of instance flags.", obj->GetEntry());
-    if (mEvents.empty() && at)
-        sLog->outErrorDb("SmartScript: AreaTrigger %u has events but no events added to list because of instance flags. NOTE: triggers can not handle any instance flags.", at->entry);
```
Comment (FrancescoBorzi) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]: why removing those?

Comment (grimgravy) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: This was removed by trinitycore. The SAI is correct, the errro information was not to be shown on the console [1496](https://github.com/azerothcore/azerothcore-wotlk/issues/1496)

Comment (FrancescoBorzi) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: @grimgravy thanks for the answer

**Comment by FrancescoBorzi** [comment]
I guess a test for this would be a quick check that BGs still work properly

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 3]
```diff
@@ -4175,7 +4175,7 @@ void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTrigger c
         if (at)
             sLog->outDebug(LOG_FILTER_DATABASE_AI, "SmartScript: EventMap for AreaTrigger %u is empty but is using SmartScript.", at->entry);
 #endif
-        return;
+        return (void)at;
```
Comment (Nefertumm) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]: Why? Just leave it as `return;`, don't cast it

**Comment by Viste** [comment]
@grimgravy use deepl.com he translates better then google

File: src/server/game/AI/SmartScripts/SmartScript.cpp (hunk 3 above)
Comment (grimgravy) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: have a better idea. Post here

Comment (Stoabrogga) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: Hmm, would be better to remove it here and add it at the beginning of the function with an appropriate comment, e.g.:
```cpp
void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTrigger const* at)
{
    (void)at; // ensure that the variable is referenced even if extra logs are disabled in order to pass compiler checks

    if (e.empty())
    {
```

**Comment by poszer** [comment]
**Tested without this commit :**
There are errors, which spam WC all the time while in the Battleground ( Alterac Valley )
Image below :
1. 
Before
2.
Before1

**Test with this commit :**
1. The modifications are applied correctly, without issue.
2. ========== Build: 5 succeeded, 0 failed, 10 up-to-date, 1 skipped ==========
3. Tested on OS: Win
4. Tested in game, everything is fine, without issues/error in World/ DDerrors.log.

**Comment by GitVerge** [comment]
This should be fixed at Battleground::GetBGCreature level to decide whether or not you want to log this error. Just avoiding this function to do exactly the same without the log doesn't feel correct.

An optional param for this function to log or not would be better (true as default) ^^

**Comment by FrancescoBorzi** [comment]
good job @grimgravy and @poszer

**Pull request merged**